To: vim_dev@googlegroups.com
Subject: Patch 7.3.396
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.396
Problem:    After forcing an operator to be characterwise it can still become
	    linewise when spanning whole lines.
Solution:   Don't make the operator linewise when motion_force was set.
	    (Christian Brabandt)
Files:	    src/ops.c


*** ../vim-7.3.395/src/ops.c	2012-01-10 13:44:23.000000000 +0100
--- src/ops.c	2012-01-10 13:28:05.000000000 +0100
***************
*** 1648,1653 ****
--- 1648,1654 ----
  	    && !oap->block_mode
  #endif
  	    && oap->line_count > 1
+ 	    && oap->motion_force == NUL
  	    && oap->op_type == OP_DELETE)
      {
  	ptr = ml_get(oap->end.lnum) + oap->end.col;
*** ../vim-7.3.395/src/version.c	2012-01-10 13:44:23.000000000 +0100
--- src/version.c	2012-01-10 13:45:31.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     396,
  /**/

-- 
Any sufficiently advanced technology is indistinguishable from magic.
					Arthur C. Clarke
Any sufficiently advanced bug is indistinguishable from a feature.
                                        Rich Kulawiec

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
